apps/web/src/pages/workflow/components/import-modal/index.tsx

```diff
@@ -22,18 +23,50 @@ const ImportModal: React.FC<Props> = ({ visible, onCancel, onUpload, ...props })
     const { control, formState, handleSubmit, reset } = useForm<FormDataProps>();
 
     const onSubmit: SubmitHandler<FormDataProps> = async ({ ...params }) => {
-        if (onUpload) {
-            await onUpload(params);
+        const { res, contains } = await validateFile(params.file[0]);
+        if (res && contains) {
+            if (onUpload) {
+                await onUpload(contains);
+            }
+            // Clear the form data upon confirmation.
+            reset();
+        } else {
+            toast.error(getIntlText('workflow.message.import_dsl_error'));
+            reset();
         }
-        // Clear the form data upon confirmation.
-        reset();
     };
 
     const handleCancel = useMemoizedFn(() => {
         reset();
         onCancel && onCancel();
     });
-
+    const validateFile = (file: File): Promise<{ res: boolean; contains?: WorkflowSchema }> => {
+        return new Promise(resolve => {
+            if (file) {
+                const reader = new FileReader();
+                reader.onload = event => {
+                    try {
+                        const result: WorkflowSchema = JSON.parse(event.target?.result as string);
+                        if (result.nodes?.length) {
+                            const nodeTypes = Object.values(basicNodeConfigs).map(
+                                item => item.type,
+                            );
+                            const isError = result.nodes.some(
+                                item => !nodeTypes.includes(item.type as WorkflowNodeType),
+                            );
+                            resolve({ res: !isError, contains: result });
+                        }
+                        resolve({ res: false });
+                    } catch (error) {
+                        resolve({ res: false });
+                    }
+                };
+                reader.readAsText(file);
+            } else {
+                resolve({ res: false });
+            }
+        });
```

Comment:
The nesting level is a bit deep, affecting maintainability, and can be optimized.